request-cache with --no-implicit-reexport cause an AssertionError

**Crash Report**

Only with `--no-implicit-reexport`, adding `from requests_cache import CachedSession` into any file, including an empty file cause this error: `AssertionError: Definition of requests_cache.patcher.BaseCache is unexpectedly incomplete`

**Traceback**

```
$ mypy --no-implicit-reexport app/test.py
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.9/bin/mypy", line 8, in <module>
    sys.exit(console_entry())
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/mypy/__main__.py", line 11, in console_entry
    main(None, sys.stdout, sys.stderr)
  File "mypy/main.py", line 87, in main
  File "mypy/main.py", line 165, in run_build
  File "mypy/build.py", line 179, in build
  File "mypy/build.py", line 254, in _build
  File "mypy/build.py", line 2697, in dispatch
  File "mypy/build.py", line 3021, in process_graph
  File "mypy/build.py", line 3138, in process_stale_scc
  File "mypy/build.py", line 2288, in write_cache
  File "mypy/build.py", line 1475, in write_cache
  File "mypy/nodes.py", line 313, in serialize
  File "mypy/nodes.py", line 3149, in serialize
  File "mypy/nodes.py", line 3083, in serialize
AssertionError: Definition of requests_cache.patcher.BaseCache is unexpectedly incomplete```
```

**To Reproduce**

```
# test.py
from requests_cache import CachedSession
```

On the command line:
```
$ mypy --no-implicit-reexport test.py
```

Note: removing the `.mypy_chache` did not help.

**Your Environment**

- request-cache used: 0.7.3
- Mypy version used: 0.910
- Mypy command-line flags: --no-implicit-reexport
- Mypy configuration options from `mypy.ini` (and other config files): not used
- Python version used: 3.9.1
- Operating system and version: macOS Version 11.3.1

--- a/mypy/semanal.py
+++ b/mypy/semanal.py
@@ -1933,7 +1933,9 @@ class SemanticAnalyzer(NodeVisitor[None],
         if self.is_incomplete_namespace(import_id):
             # We don't know whether the name will be there, since the namespace
             # is incomplete. Defer the current target.
-            self.mark_incomplete(imported_id, context)
+            self.mark_incomplete(
+                imported_id, context, module_public=module_public, module_hidden=module_hidden
+            )
             return
         message = 'Module "{}" has no attribute "{}"'.format(import_id, source_id)
         # Suggest alternatives, if any match is found.

--- a/test-data/unit/check-incremental.test
+++ b/test-data/unit/check-incremental.test
@@ -5585,3 +5585,28 @@ bar.x + 0
 x = 0
 [rechecked]
 [stale]
+
+[case testExplicitReexportImportCycleWildcard]
+# flags: --no-implicit-reexport
+import pkg.a
+[file pkg/__init__.pyi]
+
+[file pkg/a.pyi]
+MYPY = False
+if MYPY:
+    from pkg.b import B
+
+[file pkg/b.pyi]
+import pkg.a
+MYPY = False
+if MYPY:
+    from pkg.c import C
+class B:
+    pass
+
+[file pkg/c.pyi]
+from pkg.a import *
+class C:
+    pass
+[rechecked]
+[stale]

